core/src/test/java/org/jdbi/v3/core/mapper/reflect/FieldMapperPrivateMockTest.java

```diff
@@ -0,0 +1,211 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.jdbi.v3.core.mapper.reflect;
+
+import java.math.BigDecimal;
+import java.sql.ResultSet;
+import java.sql.ResultSetMetaData;
+import java.sql.SQLException;
+
+import org.jdbi.v3.core.Handle;
+import org.jdbi.v3.core.HandleAccess;
+import org.jdbi.v3.core.SamplePrivateBean;
+import org.jdbi.v3.core.ValueType;
+import org.jdbi.v3.core.mapper.RowMapper;
+import org.jdbi.v3.core.mapper.ValueTypeMapper;
+import org.jdbi.v3.core.statement.StatementContext;
+import org.jdbi.v3.core.statement.StatementContextAccess;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.mockito.junit.jupiter.MockitoSettings;
+import org.mockito.quality.Strictness;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+public class FieldMapperPrivateMockTest {
+
+    @Mock
+    ResultSet resultSet;
+
+    @Mock
+    ResultSetMetaData resultSetMetaData;
+
+    Handle handle;
+    StatementContext ctx;
+
+    RowMapper<SamplePrivateBean> mapper = FieldMapper.of(SamplePrivateBean.class);
+
+    @BeforeEach
+    public void setUpMocks() throws SQLException {
+        handle = HandleAccess.createHandle();
+        ctx = StatementContextAccess.createContext(handle);
+        when(resultSet.getMetaData()).thenReturn(resultSetMetaData);
+    }
+
+    private void mockColumns(String... columns) throws SQLException {
+        when(resultSetMetaData.getColumnCount()).thenReturn(columns.length);
+        for (int i = 0; i < columns.length; i++) {
+            when(resultSetMetaData.getColumnLabel(i + 1)).thenReturn(columns[i]);
+        }
+    }
+
+    @Test
+    public void shouldSetValueOnPrivateField() throws Exception {
+        mockColumns("longField");
+
+        Long aLongVal = 100L;
+        when(resultSet.getLong(1)).thenReturn(aLongVal);
+        when(resultSet.wasNull()).thenReturn(false);
+
+        SamplePrivateBean sampleBean = mapper.map(resultSet, ctx);
+
+        assertThat(sampleBean.getLongField()).isEqualTo(aLongVal);
+    }
+
+    @Test
+    public void shouldHandleEmptyResult() throws Exception {
```

Comment:
ironically, this test makes no sense in real life. This sets up the ResultSet to contain zero columns, which, from what I can see, is not possible in actual SQL. - https://stackoverflow.com/a/2969590/4401166

Another reason why mock tests are bad.

Reply:
and before suggesting using a join mapper where no column maps onto the FieldMapper: There is an explicit check to throw an exception if any column exists in the result set and none maps onto the fieldmapper. This may actually be a bug / limitation.